src/server/api/common/generate-replies-query.ts

```diff
@@ -1,22 +1,24 @@
 import { User } from '@/models/entities/user';
 import { Brackets, SelectQueryBuilder } from 'typeorm';
 
-export function generateRepliesQuery(q: SelectQueryBuilder<any>, me?: { id: User['id'] } | null) {
-	if (me == null) {
+export function generateRepliesQuery(q: SelectQueryBuilder<any>, me: any) {
+	const userId = me['id'] ?? null;
+	const showReplies: boolean = me['showTimelineReplies'] ?? false;
+	if (userId == null) {
 		q.andWhere(new Brackets(qb => { qb
 			.where(`note.replyId IS NULL`) // 返信ではない
 			.orWhere(new Brackets(qb => { qb // 返信だけど投稿者自身への返信
 				.where(`note.replyId IS NOT NULL`)
 				.andWhere('note.replyUserId = note.userId');
 			}));
 		}));
-	} else {
+	} else if (!showReplies) {
 		q.andWhere(new Brackets(qb => { qb
 			.where(`note.replyId IS NULL`) // 返信ではない
-			.orWhere('note.replyUserId = :meId', { meId: me.id }) // 返信だけど自分のノートへの返信
+			.orWhere('note.replyUserId = :meId', { meId: userId }) // 返信だけど自分のノートへの返信
 			.orWhere(new Brackets(qb => { qb // 返信だけど自分の行った返信
 				.where(`note.replyId IS NOT NULL`)
-				.andWhere('note.userId = :meId', { meId: me.id });
+				.andWhere('note.userId = :meId', { meId: userId });
```

Comment:
after remove `userId` and `showReplies` variables, you also need to update this codes with `me.id` and `me.showTimelineReplies` (sorry, I cant create suggestions by GitHub's limitation, it saying `Applying suggestions on deleted lines is not supported.`)